warp10/src/main/java/io/warp10/script/ext/http/HttpWarpScriptExtension.java

```diff
@@ -0,0 +1,172 @@
+//
+//   Copyright 2021  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.ext.http;
+
+import io.warp10.WarpConfig;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.warp.sdk.WarpScriptExtension;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Extension for HTTP and the associated function to change limits: MAXURLCOUNT and MAXDOWNLOADSIZE
+ */
+public class HttpWarpScriptExtension extends WarpScriptExtension {
+
+  //
+  // CAPABILITIES
+  //
+
+  /**
+   * If this configuration is set, then HTTP function will check for this capability.
+   */
+  public static final String HTTP_CAPABILITY = "warpscript.http.capability";
+
+  //
+  // CONFIGURATION
+  //
+
+  /**
+   * Maximum number of calls to HTTP in a session
+   */
+  public static final String WARPSCRIPT_HTTP_LIMIT = "warpscript.http.limit";
+  public static final String WARPSCRIPT_HTTP_LIMIT_HARD = "warpscript.http.limit.hard";
+
+  /**
+   * Maximum cumulative size of content retrieved via calls to HTTP in a session
+   */
+  public static final String WARPSCRIPT_HTTP_MAXSIZE = "warpscript.http.maxsize";
+  public static final String WARPSCRIPT_HTTP_MAXSIZE_HARD = "warpscript.http.maxsize.hard";
+
+  /**
+   * Allowed and excluded host patterns.
+   */
+  public static final String WARPSCRIPT_HTTP_HOST_PATTERNS = "warpscript.http.host.patterns";
+
+  //
+  // STACK
+  //
+
+  /**
+   * Maximum number of calls to HTTP in a session
+   */
+  public static final String ATTRIBUTE_HTTP_LIMIT = "http.limit";
+  public static final String ATTRIBUTE_HTTP_LIMIT_HARD = "http.limit.hard";
+
+  /**
+   * Number of calls to HTTP so far in the sessions
+   */
+  public static final String ATTRIBUTE_HTTP_COUNT = "http.count";
+
+  /**
+   * Maximum cumulative size of content retrieved via calls to HTTP in a session
+   */
+  public static final String ATTRIBUTE_HTTP_MAXSIZE = "http.maxsize";
+  public static final String ATTRIBUTE_HTTP_MAXSIZE_HARD = "http.maxsize.hard";
+
+  /**
+   * Current  HTTP so far in the sessions
+   */
+  public static final String ATTRIBUTE_HTTP_SIZE = "http.size";
+
+  //
+  // DEFAULTS
+  //
+
+  public static final long DEFAULT_HTTP_LIMIT = 64L;
```

Comment:
this is too high, limit to 1 so users explicitly set the configuration.

Reply:
Limit has been set to 1L. But since auth_macro may need to make requests as well (e.g. digest authentication), should we raise this default to a few more ?